### ISSUE TITLE
error popup still off screen sometimes in UI

### ISSUE THREAD

**Issue by erg** [factor/factor#115]
Simply adding a scroller will not fully fix this issue. We might want to see if the popup has more room above or below and show it there.

Can't see all the area, `<scroller>` helps here:

<img width="409" alt="Screen Shot 2021-12-21 at 8 36 37 PM" src="https://user-images.githubusercontent.com/27430/147025625-e37d1d32-cea1-40d0-a4fd-56497e046a0e.png">

Can't see any of the area, showing above instead of below would help:

<img width="413" alt="Screen Shot 2021-12-21 at 8 37 07 PM" src="https://user-images.githubusercontent.com/27430/147025687-c7e53ada-b3ae-41ff-940a-9674fa6dec87.png">

**Comment by erg** [comment]
If we add a scrollbar to the popup it would at least be reachable/clickable. We have the same problem with selecting an editor from the restarts; it's offscreen and not scrollable.

**Comment by erg** [comment]
Here's a possible patch, but maybe the scroller should be somewhere else:

```
diff --git a/basis/ui/tools/listener/listener.factor b/basis/ui/tools/listener/listener.factor
index 98c0a0fae9..bc7a1b700b 100644
--- a/basis/ui/tools/listener/listener.factor
+++ b/basis/ui/tools/listener/listener.factor
@@ -347,7 +347,7 @@ M: object accept-completion-hook 2drop ;
     ] ;
 
 : frame-debugger ( debugger -- labeled )
-    "Error" debugger-color <framed-labeled-gadget> ;
+    "Error" debugger-color <framed-labeled-gadget> <scroller> ;
 
 :: <debugger-popup> ( error continuation interactor -- popup )
     error
```

**Comment by mrjbq7** [comment]
Fixed in d023a4b7e24fe44741b8509f0c83ac9ed59b877d.
